eras/dijkstra/src/Cardano/Ledger/Dijkstra/TxCert.hs

```diff
@@ -1,135 +1,344 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE UndecidableSuperClasses #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
 
-module Cardano.Ledger.Dijkstra.TxCert () where
-
+module Cardano.Ledger.Dijkstra.TxCert (
+  DijkstraTxCertUpgradeError,
+  DijkstraTxCert (..),
+  DijkstraDelegCert (..),
+) where
+
+import Cardano.Ledger.BaseTypes (kindObject)
+import Cardano.Ledger.Binary (
+  DecCBOR (..),
+  Decoder,
+  EncCBOR (..),
+  FromCBOR (..),
+  ToCBOR (..),
+  decodeRecordSum,
+  encodeListLen,
+  encodeWord8,
+  invalidKey,
+  toPlainEncoding,
+ )
+import Cardano.Ledger.Coin (Coin)
 import Cardano.Ledger.Conway.Core (
   ConwayEraTxCert,
-  ShelleyEraTxCert (..),
   notSupportedInThisEra,
+  poolCertKeyHashWitness,
+  pattern AuthCommitteeHotKeyTxCert,
+  pattern RegDRepTxCert,
   pattern RegDepositDelegTxCert,
   pattern RegDepositTxCert,
-  pattern RegTxCert,
+  pattern RegPoolTxCert,
+  pattern ResignCommitteeColdTxCert,
+  pattern RetirePoolTxCert,
+  pattern UnRegDRepTxCert,
   pattern UnRegDepositTxCert,
-  pattern UnRegTxCert,
+  pattern UpdateDRepTxCert,
  )
 import Cardano.Ledger.Conway.TxCert (
-  ConwayDelegCert (..),
   ConwayEraTxCert (..),
   ConwayGovCert (..),
-  ConwayTxCert (..),
+  Delegatee (..),
   conwayTotalDepositsTxCerts,
   conwayTotalRefundsTxCerts,
-  getScriptWitnessConwayTxCert,
-  getVKeyWitnessConwayTxCert,
-  pattern ConwayRegCert,
-  pattern DelegStake,
+  conwayTxCertDelegDecoder,
+ )
+import Cardano.Ledger.Core (
+  Era,
+  EraTxCert (..),
+  KeyHash,
+  KeyRole (..),
+  PoolCert (..),
+  ScriptHash,
+  Value,
+  eraProtVerLow,
+  fromEraCBOR,
  )
-import Cardano.Ledger.Core (EraTxCert (..), PoolCert (..))
+import Cardano.Ledger.Credential (StakeCredential, credKeyHashWitness, credScriptHash)
 import Cardano.Ledger.Dijkstra.Era (DijkstraEra)
 import Cardano.Ledger.Dijkstra.PParams ()
-import Data.Coerce (coerce)
-import Data.Maybe.Strict (StrictMaybe (..))
-import Data.Void (Void)
+import Cardano.Ledger.Shelley.TxCert (encodePoolCert, poolTxCertDecoder)
+import Cardano.Ledger.Val (Val)
+import Control.DeepSeq (NFData)
+import Data.Aeson (KeyValue ((.=)), ToJSON (..))
+import GHC.Generics (Generic)
+import NoThunks.Class (NoThunks)
+
+data DijkstraDelegCert
+  = DijkstraRegCert !StakeCredential !Coin
+  | DijkstraUnRegCert !StakeCredential !Coin
+  | DijkstraRegDelegCert !StakeCredential !Delegatee !Coin
```

Comment:
We need to preserve this certificate, otherwise people will not be change their delegation without unregistration
```suggestion
  | DijkstraDelegCert !StakeCredential !Delegatee
  | DijkstraRegDelegCert !StakeCredential !Delegatee !Coin
```